polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -603,36 +702,60 @@ impl<H: Copy> CandidateDescriptorV2<H> {
 }
 
 impl<H: Copy> CommittedCandidateReceiptV2<H> {
-	/// Checks if descriptor core index is equal to the committed core index.
-	/// Input `cores_per_para` is a claim queue snapshot at the candidate's relay parent, stored as
+	/// Performs checks on the UMP signals and returns them.
+	///
+	/// Also checks if descriptor core index is equal to the committed core index.
+	///
+	/// Params:
+	/// - `cores_per_para` is a claim queue snapshot at the candidate's relay parent, stored as
 	/// a mapping between `ParaId` and the cores assigned per depth.
-	pub fn check_core_index(
+	/// - `allow_approved_peer_signal` should be true if the `ApprovedPeerUmpSignal` node feature
+	///   bit is enabled. If it is not, `ApprovedPeer` UMP signals are not allowed.
+	pub fn check_ump_signals(
```

Comment:
Why do you need the allowed_approved_peer_signal here ?

Reply:
Because before the feature is enabled, the code should forbid more than one ump signals.
Otherwise, some validators that are upgraded would allow two ump signals whereas unupgraded validators would only allow the core selector. Leading to disputes and innocent validators being slashed. The same reason we used a node feature for v2 receipts